Small changes and possible bug fixes

balaram26 opened the pull request:
Changes have been made on: fit_p_value,compute_ig and compute_tuples. Check the commit log for detailed information.

balaram26 committed:
test prints addeded for running test analysis

bala-hedev committed:
Small changes done

1.fit_p_value cpp modified for contrast data check.
2.division parameter included in compute tuples.
3.Compute max ig discrete function fixed.(had issue with variable initiation).

yoctozepto commented on .vscode/c_cpp_properties.json:
```diff
@@ -0,0 +1,19 @@
+{
```
We don't want to commit this.

yoctozepto commented on .gitignore:
```diff
@@ -4,3 +4,14 @@ __pycache__
 # Python build/setuptools
 dist/
 *.egg-info/
+build/*
```
Why are these extra ignores needed? We should not need them. They feel arbitrary.

balaram26 replied:
I have some test data and also some example scripts on my local system. i didn't want them to be committed at that stage as some are just test. once its fully done, I might add them as example test scripts in python, so I let them b ignored for now.

yoctozepto commented on data/MDFS_Result_python_1D.json:
```diff
@@ -172,41 +172,41 @@
         3.3997185080709654e-07,
         0.9754512271097073,
-        0.20721572256977,
+        0.20721572256977003,
```
Any idea why these values have changed?

balaram26 replied:
no, i dont remember working with this file. so not sure. i saw somehow the file was modified, was not sure. we can ignore these changes.

yoctozepto commented on example_mdfs_run.json:
```diff
@@ -0,0 +1 @@
+{"contrast_indices": 
```
Some broken file.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -97,31 +97,44 @@ def compute_max_ig(data, decision, contrast_data=None, dimensions=1, divisions=1
     iv_count = 0 if interesting_vars is None else len(interesting_vars)
     in_interesting = (c_int * iv_count)(*interesting_vars) if interesting_vars else (c_int * iv_count)()
 
+
```
Needless line.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -97,31 +97,44 @@ def compute_max_ig(data, decision, contrast_data=None, dimensions=1, divisions=1
     else:
         n_contrast_variables = 0
-
+        
```
Needless ws.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -97,31 +97,44 @@ def compute_max_ig(data, decision, contrast_data=None, dimensions=1, divisions=1
-    seed = -1 if seed is None else seed
-    range_ = -1 if range_ is None else range_
+    seed = -1
+    range_ = -1
```
Good catch. These two lines should have been dropped.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -132,8 +145,15 @@ def compute_max_ig_discrete(data, decision, contrast_data=None, dimensions=1, pc
     else:
         n_contrast_variables = 0
-
+    
```
Needless ws.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -132,8 +145,15 @@ def compute_max_ig_discrete(data, decision, contrast_data=None, dimensions=1, pc
+    #     unique_values = np.unique(data)
+    #     divisions = len(unique_values) - 1
+    #     divisions = prepare_integer_in_bounds(divisions,"Divisions",1,15)
+
```
Needless line and code-comment.

balaram26 replied:
i had left those in case we wanted to apply some division-related operations, they were a replication of the R-code.

yoctozepto commented on mdfs/compute_max_ig.py:
```diff
@@ -97,31 +97,44 @@ def compute_max_ig(data, decision, contrast_data=None, dimensions=1, divisions=1
 
     return handle_error(result)
 
+def prepare_integer_in_bounds(x, x_desc, lower_bound, upper_bound):
```
Not used.

balaram26 replied:
Again, it was division related code, implementation of R-code function. kept them as optional if you wanted to apply division-related suggestions. We can remove those now I guess.

yoctozepto commented on mdfs/compute_tuples.py:
```diff
@@ -145,7 +145,7 @@ def compute_tuples(data, decision=None, *, dimensions=2, divisions=1,
 
 
-def compute_tuples_discrete(data, decision=None, *, dimensions=2, pc_xi=0.25,
+def compute_tuples_discrete(data, decision=None, *, divisions = 1, dimensions=2, pc_xi=0.25,
```
Should not be here (discussed already).

yoctozepto commented on mdfs/compute_tuples.py:
```diff
@@ -172,7 +172,7 @@ def compute_tuples_discrete(data, decision=None, *, dimensions=2, pc_xi=0.25,
             raise Exception("i_lower should have the length of variables count.")
 
-    divisions = len(set(data)) - 1
+    # divisions = len(set(data)) - 1
```
This should stay as is for now. We might want to add more user-friendliness in here but it's not mandatory since we want to reimplement this anyway.

yoctozepto commented on mdfs/fit_p_value.py:
```diff
@@ -43,13 +43,17 @@ def __del__(self):
 
 def fit_p_value(chisq, chisq_contrast, *,
-                exponential_fit,
+                exponential_fit=None,
```
It does not make sense to have a default here. It was on purpose.

yoctozepto commented on mdfs/fit_p_value.py:
```diff
@@ -43,13 +43,17 @@ def __del__(self):
     n_vars = len(chisq)
-    n_contrasts = len(chisq_contrast)
+
+    if chisq_contrast!= None:
```
We don't need to play with it. The error handling has been added by you in C++ code and it's enough.

yoctozepto commented on mdfs/run.py:
```diff
@@ -57,7 +57,7 @@ def run(data, decision, *, n_contrast=None, dimensions=1, divisions=1, discretiz
                                  exponential_fit=(fit_mode=="exp"))
         mdfs_result["p_value"] = fpv_result.p_values.copy()
-
+    
```
Needless ws.

yoctozepto commented on setup.py:
```diff
@@ -53,6 +53,7 @@ def get_tag(self):
                 "shared/src/cpu/common.cpp",
                 "shared/src/cpu/discretize.cpp",
             ],
+            extra_compile_args=['-std=c++14', '-Wall', '-Wextra'],  # Add your compiler flags here
```
Why so?

yoctozepto commented on shared/interface/fit_p_value.cpp:
```diff
@@ -1,5 +1,4 @@
 #include "common.h"
-
```
This line was separating custom headers from system headers for readability.

yoctozepto commented on shared/src/cpu/PrintUtility.h:
```diff
@@ -0,0 +1,48 @@
+#ifndef PRINT_UTILITY_H
```
Why this? It's not used.

balaram26 replied:
It was my handy file to print things and see data at every step for testing. Thought I removed it at last before committing the files.

yoctozepto commented:
@balaram26 Are you planning to update this PR?

balaram26 commented:
Hey i have addressed all issues you asked here, do I need to update code on my side or you can just not accept those not needed changes and do merging for accepted one alone? this way I can fetch and then do next pull after that? @yoctozepto
